config/routes.rb

```diff
@@ -15,7 +15,7 @@
   get 'settings' => 'users#settings', as: 'settings'
   get 'help' => 'pages#help', as: 'help'
   get 'activity' => 'pages#activity_log', as: 'activity_log'
-  post 'activity/filter' => 'pages#activity_log_filter', as: 'activity_log_filter'
```

Comment:
This route was not matching up with the POST request generated in `_filtering.js`, so all filters were resulting in a 404 error.